src/sensor.cpp

```diff
@@ -628,6 +628,18 @@ void log_callback_end( uint32_t fps,
         raise_on_before_streaming_changes(true); //Required to be just before actual start allow recording to work
         _source.set_callback(callback);
         _is_streaming = true;
+
+        std::string device_name = this->get_device().shared_from_this()->get_info(RS2_CAMERA_INFO_NAME);
+        auto active_streams = get_active_streams();
```

Comment:
Again too much code.

Try following the 1 line injection :)
It doesn't make sense that 1-2 line function will be 10 times larger because we want to add AUS.
We want as minimal changes as possible and the logic at a dedicated place.

Reply:
?

Reply:
changed